apps/backend/src/modules/routes/route.service.ts

```diff
@@ -84,7 +84,7 @@ class RouteService implements Service {
 			pois: payload.poiIds.map((id, index) => ({
 				id,
 				visitOrder: index,
-			})),
+			})) as { id: number; name: string; visitOrder: number }[],
```

Comment:
What I meant by the previous comment is that there shouldn't be any mention of the name in initalizeNew for the routeEntity. It means it should expect the interface without name in the params and it should return the interface without a name. Is there an issue with this? 
```suggestion
			}))
```

Reply:
@liza-veis initializeNew returns RouteEntity and the name field in pois is required and then an error

Reply:
I think there was a slight misunderstanding. What we need to do is to make it optional on entity level, and eliminate it for initializeNew and toNewObject object methods, while keeping it as required in toListObject and toObject methods (we will need to cast their types for this). What we want to achieve, is just to be able to use initializeNew and toNewObject without name, while all other methods should have name as required. In other places, like model and response dto, we still want to keep name as required